app/main.py

```diff
@@ -1,3 +1,58 @@
+from typing import Union
+from functools import total_ordering
+
+
+@total_ordering
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
```

Comment:
It's good that you're checking the type of 'other' before performing mathematical operations. However, it would be more efficient to use the built-in function 'numbers.Real' instead of checking against both 'int' and 'float'. This would also allow your code to handle any other types of numbers that might be added in the future.